sympy/combinatorics/coset_table.py

```diff
@@ -151,9 +151,18 @@ def define(self, alpha, x):
         table.append([None]*len(A))
         # beta is the new coset generated
         beta = len_table
+        if modified:
+            H = self.subgroup
+            self._grp = free_group(', ' .join(["a_%d" % i for i in range(len(H))]))[0]
+            self.P.append([None]*len(self.A))
+            self.p_p = []
```

Comment:
All of this should still be done outside of this function, in the main `coset_enumeration` - it only needs doing once but `define` is called a lot of times...